apis/python/src/tiledb/vector_search/ingestion.py

```diff
@@ -2137,25 +2147,70 @@ def create_ingestion_dag(
             input_vectors_work_items_per_worker_during_sampling
         )
 
+        def scale_resources(
+            min_resource_ram,
+            max_resource_ram,
+            min_resource_cpu,
+            max_resource_cpu,
+            max_input_size,
+            input_size,
+        ):
+            cpu = str(
+                max(
+                    min(max_resource_cpu, threads * input_size / max_input_size),
+                    min_resource_cpu,
+                )
+            )
+            ram = (
+                str(
+                    max(
+                        min_resource_ram,
+                        min(
+                            max_resource_ram,
+                            int(max_resource_ram * input_size / max_input_size),
+                        ),
+                    )
+                )
+                + "Gi"
+            )
+            return {"cpu": cpu, "memory": ram}
+
         # We can't set as default in the function due to the use of `str(threads)`
         # For consistency we then apply all defaults for resources here.
         if ingest_resources is None:
-            ingest_resources = {"cpu": str(threads), "memory": "16Gi"}
+            ingest_resources = scale_resources(
+                2, 16, 2, 16, DEFAULT_PARTITION_BYTE_SIZE, size_in_bytes
+            )
 
         if consolidate_partition_resources is None:
-            consolidate_partition_resources = {"cpu": str(threads), "memory": "16Gi"}
+            consolidate_partition_resources = scale_resources(
+                2, 16, 2, 16, DEFAULT_PARTITION_BYTE_SIZE, size_in_bytes
+            )
 
         if copy_centroids_resources is None:
             copy_centroids_resources = {"cpu": "1", "memory": "2Gi"}
 
         if random_sample_resources is None:
-            random_sample_resources = {"cpu": "2", "memory": "6Gi"}
+            random_sample_resources = {
+                "cpu": "2",
+                "memory": "6Gi",
+            }
 
         if kmeans_resources is None:
-            kmeans_resources = {"cpu": "8", "memory": "32Gi"}
+            kmeans_resources = scale_resources(
```

Comment:
I would prefer to have 2 separate `scale_resource` function calls for CPU and memory in order to be more clear which `min,max` values apply to which resource.

Reply:
Changed it